File: src/p0003/cpp/solution.cpp
class Solution {
public:
    int lengthOfLongestSubstring(const string &s) {
        vector<int> count(256, 0);
        size_t result = 0, i = 0, j = 0;
        while (j < s.length()) {
            count[s[j]] += 1;
            while (count[s[j]] > 1) {
                count[s[i]] -= 1;
                i += 1;
            }
            result = max(result, j + 1 - i);
            j += 1;
        }
        return result;
    }
};

File: src/p0003/golang/solution.go
func lengthOfLongestSubstring(s string) int {
	count := make([]int, 256)
	i, j, result := 0, 0, 0
	for j < len(s) {
		count[s[j]] += 1
		for count[s[j]] > 1 {
			count[s[i]] -= 1
			i += 1
		}
		if result < j+1-i {
			result = j + 1 - i
		}
		j += 1
	}
	return result
}
